scripts/cc-cli/src/utils/tx.ts

```diff
@@ -0,0 +1,67 @@
+import { ISubmittableResult } from "@polkadot/types/types";
+import { ApiPromise, KeyringPair } from "creditcoin-js";
+
+import { SubmittableExtrinsic } from "@polkadot/api/types";
+
+export async function signSendAndWatch(
+  tx: SubmittableExtrinsic<"promise", ISubmittableResult>,
+  api: ApiPromise,
+  signer: KeyringPair
+): Promise<TxResult> {
+  // Top-Level Promise
+
+  // eslint-disable-next-line @typescript-eslint/no-misused-promises
+  return new Promise(async (resolve) => {
+    // eslint-disable-next-line @typescript-eslint/no-misused-promises
+    const out: TxResult = await new Promise(async (resolveInner) => {
+      console.log("Sending transaction...");
+      // Sign and send with callback
+      await tx.signAndSend(signer, ({ status, dispatchError }) => {
```

Comment:
You should also unsubscribe when you're about to resolve, AFAIK otherwise these subscriptions will stay open.